<a href="https://colab.research.google.com/github/Ksumeyye/Ucus_Rezervasyon_Sistemi/blob/master/sumeyye_kaya_numpy1_alistirmalar.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Alıştırmalar**
1) Manuel değerleri el ile girerek 3 boyutlu bir array oluşturup bir değişkene atayın. Ardından 3 boyutlu olup olmadığına bakmak için dimension'ını kontrol edin
2) 34, 40, 46, 52... 112 şeklinde devam eden 1 boyutlu bir array oluşturun.
3) 50-500 arasında lineer artış gösteren 91 tane tam sayıdan oluşan bir array oluşturun.(dtype'ı int olsun)
4) 100(10**2) ile 10000(10**5) arasında logaritmik artış gösteren 8 sayıdan oluşan bir array oluşturun.
5) 0-8 tam arasındaki ardışık tam sayılardan oluşan(0 ve 8 dahil toplam 9 değer) 3x3 shape'inde bir matris oluşturun.
6) 6x6 formatında bir sıfır matrisi oluşturun. (dtype'ı int olsun)
7) 4x4 formatında bir bir matrisi oluşturun. (dtype'ı int olsun)
8) 8x8 formatında bir birim matris oluşturun. (Sadece sol köşegeni 1 geri kalan değerleri 0 olan matris'e birim matris deniyor.) (dtype'ı int olsun)
9) 5x5 formatında bir köşegen matrisi oluşturun (Sadece sol üstten sağ alta doğru olan köşegendeki değerleri 3 olsun diğer bütün değerleri 0) (Bunu henüz görmediniz ama birim matrise benziyor, sadece köşegen değerleri 1 değil 3 olacak. Bir şeyler düşünün)
10) np.random modülünden uygun fonksiyonu kullanarak 0-1 arasında toplam 6 tane değerden oluşan 1 boyutlu bir array oluşturun.
11) np.random modülünden uygun fonksiyonu kullanarak 50-100 arasındaki(50 ve 100 dahil) tam sayılardan 10 tanesiyle oluşan (5,2) shape'inde bir array oluşturun. Ardından bu arrayin shape'ini kontrol edin.
12) np.random modülünden uygun fonksiyonu kullanarak 100-1000(1000 dahil değil) arasındaki tam sayılardan rastgele 50 tanesinden oluşan (2,5,5) shape'inde 3 boyutlu bir array oluşturun. Ardından bu array'in dimension'ını(boyutunu) ve shape'ini kontrol edin.
13) np.random modülünden uygun fonksiyonu kullanarak 0-100(0 ve 100 dahil) arasındaki tam sayılardan 10 tane seçerek bir array oluşturun. Bu array'in maximum, mininmum değerlerine ve bu değerlerin indexlerine bakın.
14) np.random modülünden uygun fonksiyonu kullanarak 300-500(300 ve 500 dahil) arasındaki tam sayılardan 20 tane seçerek (2,2,5) shape'inde 3 boyutlu bir array oluşturun. Ardından bu array'in içindeki 20 tam sayı arasından maximum ve minimum değerleri manuel olarak tespit edin ve indexleme yaparak çekmeye çalışın.
15) 0-50(50 dahil) arasındaki ardışık tam sayılardan oluşan bir array oluşturun. Ardından bu arrayin 20. ve 35. indexleri arasındaki değerleri 500'e eşitleyin ve yeni oluşan array'i ekrana yazdırarak broadcasting işleminin yapılıp yapılmadığını test edin.


In [2]:
import numpy as np

1) Manuel değerleri el ile girerek 3 boyutlu bir array oluşturup bir değişkene atayın. Ardından 3 boyutlu olup olmadığına bakmak için dimension'ını kontrol edin

In [ ]:
arr3d = np.array([[[0,1,2] , [3,4,5]] , [[6,7,8] , [9,10,11]]])
arr3d

array([[[ 0,  1,  2],
        [ 3,  4,  5]],

       [[ 6,  7,  8],
        [ 9, 10, 11]]])

In [ ]:
arr3d.ndim

3

2) 34, 40, 46, 52... 112 şeklinde devam eden 1 boyutlu bir array oluşturun. 

In [ ]:
np.arange(34,113,6)

array([ 34,  40,  46,  52,  58,  64,  70,  76,  82,  88,  94, 100, 106,
       112])

3) 50-500 arasında lineer artış gösteren 91 tane tam sayıdan oluşan bir array oluşturun.(dtype'ı int olsun) 

In [ ]:
np.linspace(50,500,91 , dtype=int)

array([ 50,  55,  60,  65,  70,  75,  80,  85,  90,  95, 100, 105, 110,
       115, 120, 125, 130, 135, 140, 145, 150, 155, 160, 165, 170, 175,
       180, 185, 190, 195, 200, 205, 210, 215, 220, 225, 230, 235, 240,
       245, 250, 255, 260, 265, 270, 275, 280, 285, 290, 295, 300, 305,
       310, 315, 320, 325, 330, 335, 340, 345, 350, 355, 360, 365, 370,
       375, 380, 385, 390, 395, 400, 405, 410, 415, 420, 425, 430, 435,
       440, 445, 450, 455, 460, 465, 470, 475, 480, 485, 490, 495, 500])

4) 100(102) ile 10000(105) arasında logaritmik artış gösteren 8 sayıdan oluşan bir array oluşturun.

In [ ]:
x = np.linspace(100,10000,8)
x

array([  100.        ,  1514.28571429,  2928.57142857,  4342.85714286,
        5757.14285714,  7171.42857143,  8585.71428571, 10000.        ])

5) 0-8 tam arasındaki ardışık tam sayılardan oluşan(0 ve 8 dahil toplam 9 değer) 3x3 shape'inde bir matris oluşturun.

In [5]:
np.random.randint(0,9,(3,3))

array([[3, 7, 8],
       [7, 2, 7],
       [4, 7, 4]])

6) 6x6 formatında bir sıfır matrisi oluşturun. (dtype'ı int olsun) 

In [6]:
np.zeros((6,6),dtype=int)

array([[0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0]])

7) 4x4 formatında bir bir matrisi oluşturun. (dtype'ı int olsun) 

In [8]:
np.random.randint(0,65,(4,4),dtype=int)

array([[20,  5, 30, 32],
       [19, 28, 52, 31],
       [ 5,  2, 57, 35],
       [ 8, 55, 31, 50]])

 8) 8x8 formatında bir birim matris oluşturun. (Sadece sol köşegeni 1 geri kalan değerleri 0 olan matris'e birim matris deniyor.) (dtype'ı int olsun) 

In [9]:
np.eye(8,dtype='int')

array([[1, 0, 0, 0, 0, 0, 0, 0],
       [0, 1, 0, 0, 0, 0, 0, 0],
       [0, 0, 1, 0, 0, 0, 0, 0],
       [0, 0, 0, 1, 0, 0, 0, 0],
       [0, 0, 0, 0, 1, 0, 0, 0],
       [0, 0, 0, 0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0, 0, 1, 0],
       [0, 0, 0, 0, 0, 0, 0, 1]])

 9) 5x5 formatında bir köşegen matrisi oluşturun (Sadece sol üstten sağ alta doğru olan köşegendeki değerleri 3 olsun diğer bütün değerleri 0) (Bunu henüz görmediniz ama birim matrise benziyor, sadece köşegen değerleri 1 değil 3 olacak. Bir şeyler düşünün)

In [13]:
np.eye(5,dtype='int')*3

array([[3, 0, 0, 0, 0],
       [0, 3, 0, 0, 0],
       [0, 0, 3, 0, 0],
       [0, 0, 0, 3, 0],
       [0, 0, 0, 0, 3]])

10) np.random modülünden uygun fonksiyonu kullanarak 0-1 arasında toplam 6 tane değerden oluşan 1 boyutlu bir array oluşturun.

In [14]:
np.random.rand(6)

array([0.67592906, 0.95279391, 0.59359478, 0.03711632, 0.56251999,
       0.24451804])

11) np.random modülünden uygun fonksiyonu kullanarak 50-100 arasındaki(50 ve 100 dahil) tam sayılardan 10 tanesiyle oluşan (5,2) shape'inde bir array oluşturun. Ardından bu arrayin shape'ini kontrol edin.

In [37]:
arr2d=np.random.randint(50,101,(5,2))
arr2d

array([[99, 55],
       [74, 62],
       [62, 86],
       [68, 69],
       [54, 93]])

In [38]:
arr2d.ndim

2

12) np.random modülünden uygun fonksiyonu kullanarak 100-1000(1000 dahil değil) arasındaki tam sayılardan rastgele 50 tanesinden oluşan (2,5,5) shape'inde 3 boyutlu bir array oluşturun. Ardından bu array'in dimension'ını(boyutunu) ve shape'ini kontrol edin.

In [48]:
arr3d=np.random.randint(100,1000,(2,5,5))
arr3d

array([[[294, 296, 760, 621, 583],
        [708, 242, 913, 193, 747],
        [744, 252, 955, 353, 103],
        [427, 646, 810, 954, 419],
        [559, 685, 907, 259, 569]],

       [[833, 820, 895, 593, 811],
        [620, 674, 636, 350, 753],
        [450, 516, 578, 624, 861],
        [454, 507, 247, 651, 776],
        [997, 766, 218, 542, 595]]])

In [49]:
arr3d.ndim

3

In [50]:
arr3d.shape

(2, 5, 5)

13) np.random modülünden uygun fonksiyonu kullanarak 0-100(0 ve 100 dahil) arasındaki tam sayılardan 10 tane seçerek bir array oluşturun. Bu array'in maximum, mininmum değerlerine ve bu değerlerin indexlerine bakın. 

In [51]:
arr=np.random.randint(0,101,10)
arr

array([86, 50, 37, 71, 12, 29, 73, 11, 32, 61])

In [52]:
arr.max()

86

In [56]:
arr.min()

11

In [54]:
arr.argmax()

0

In [55]:
arr.argmin()

7

14) np.random modülünden uygun fonksiyonu kullanarak 300-500(300 ve 500 dahil) arasındaki tam sayılardan 20 tane seçerek (2,2,5) shape'inde 3 boyutlu bir array oluşturun. Ardından bu array'in içindeki 20 tam sayı arasından maximum ve minimum değerleri manuel olarak tespit edin ve indexleme yaparak çekmeye çalışın.

In [57]:
arr3d=np.random.randint(300,501,(2,2,5))
arr3d

array([[[383, 447, 377, 414, 300],
        [349, 361, 343, 404, 357]],

       [[481, 476, 367, 496, 423],
        [307, 417, 337, 477, 456]]])

In [58]:
arr3d.min()

300

In [59]:
arr3d.max()

496

In [60]:
arr3d.argmax()

13

In [61]:
arr3d.argmin()

4

15) 0-50(50 dahil) arasındaki ardışık tam sayılardan oluşan bir array oluşturun. Ardından bu arrayin 20. ve 35. indexleri arasındaki değerleri 500'e eşitleyin ve yeni oluşan array'i ekrana yazdırarak broadcasting işleminin yapılıp yapılmadığını test edin.

In [46]:
arr=np.arange(0,51)
arr

array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50])

In [47]:
arr[20:35] = 500
arr

array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19, 500, 500, 500, 500, 500, 500,
       500, 500, 500, 500, 500, 500, 500, 500, 500,  35,  36,  37,  38,
        39,  40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50])